src/Tasks.UnitTests/Hash_Tests.cs

```diff
@@ -43,6 +43,100 @@ public void HashTaskEmptyInputTest()
             Assert.Null(zeroLengthItemsHash);
         }
 
+        [Fact]
+        public void HashTaskLargeImputCountTest()
+        {
+            // This hash was pre-computed. If the implementation changes it may need to be adjusted.
+            var expectedHash = "8a996bbcb5e481981c2fba7ac408e20d0b4360a5";
+
+            ITaskItem[] itemsToHash = new ITaskItem[1000];
+            for (int i = 0; i < itemsToHash.Length; i++)
+            {
+                itemsToHash[i] = new TaskItem($"Item{i}");
+            }
+
+            var actualHash = ExecuteHashTask(itemsToHash);
+            Assert.Equal(expectedHash, actualHash);
+        }
+
+        [Fact]
+        public void HashTaskLargeImputSizeTest()
+        {
+            // This hash was pre-computed. If the implementation changes it may need to be adjusted.
+            var expectedHash = "0509142dd3d3a733f30a52a0eec37cd727d46122";
+
+            string[] array = new string[1000];
+            for (int i = 0; i < array.Length; i++)
+            {
+                array[i] = $"Item{i}";
+            }
+            ITaskItem[] itemsToHash = new ITaskItem[] { new TaskItem(string.Join("", array)) };
+
+            var actualHash = ExecuteHashTask(itemsToHash);
+            Assert.Equal(expectedHash, actualHash);
+        }
+
+        [Fact]
+        public void HashTaskLargeImputCountAndSizeTest()
+        {
+            // This hash was pre-computed. If the implementation changes it may need to be adjusted.
+            var expectedHash = "be23ae7b09f1e14fa5a17de87ddae2c3ec62f967";
+
+            int inputSize = 1000;
+            string[][] array = new string[inputSize][];
+            for (int i = 0; i < array.Length; i++)
+            {
+                array[i] = new string[inputSize];
+                for (int j = 0; j < array[i].Length; j++)
+                {
+                    array[i][j] = $"Item{i}{j}";
```

Comment:
This is 1_000_000 iterations, is it? What is usual duration if this unit test?

Reply:
It's 1 sec. It is less than the HashTaskDifferentImputSizesTest takes.

Reply:
The duration of these tests worries me though.

Reply:
1 sec is too much, IMHO, can we get it under 100ms?

Reply:
This one is most unnecessary test of all. Maybe drop it at all?

Reply:
I would like to save the test with different sizes of input (`HashTaskDifferentInputSizesTest`) despite it takes 3 sec, cause it tests a dangerous place where one can easily make a mistake.

Reply:
+1 on trying to make the tests fast. I'd say 3 seconds is too much.